android/library/src/main/java/com/polidea/multiplatformbleadapter/BleModule.java

```diff
@@ -377,7 +377,7 @@ public void getKnownDevices(String[] deviceIdentifiers,
             }
         }
 
-        onSuccessCallback.onSuccess(knownDevices.toArray(new Device[knownDevices.size()]));
+        onSuccessCallback.onSuccess(knownDevices.toArray(new Device[0]));
```

Comment:
I don't think this change is improvement. `new Device[0]` array provided as an argument will be discarded by `toArray` method if `knownDevices` has 1 or more elements. Previous implementation made sure that allocated array will fit all of the List elements and thus will not be discarded by `toArray`.